base_ical/controllers/main.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Hunki Enterprises BV
+# Copyright 2024 initOS GmbH
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+import logging
+
+import werkzeug.wrappers
+
+from odoo import http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class ICalController(http.Controller):
+    @http.route(
+        "/base_ical/<int:calendar_id>",
+        auth="none",
+        csrf=False,
+        methods=["GET"],
+        type="http",
+    )
+    def get_ical(self, calendar_id, access_token=None):
+        if not access_token:
+            raise request.not_found()
+
+        user_id = request.env["res.users.apikeys"]._check_credentials(
+            scope=f"odoo.plugin.ical.{calendar_id}",
+            key=access_token,
+        )
+        if not user_id:
+            raise request.not_found()
+
+        calendar = (
+            http.request.env["base.ical"]
+            .sudo()
+            .search([("id", "=", calendar_id)])
+            .with_user(user_id)
+        )
+        if not calendar:
+            return request.not_found()
+
+        records = calendar._get_items()
```

Comment:
My suggestion above means that this call to `_get_items()` is no longer in sudo() mode and could return fewer results. I'm open to an argument on this.

Reply:
Before is wasn't su too. because as @hbrunn mention `with_user` sets the su flag to false per default. Either way I guess that without sudo would be correct because you don't want to allow access to stuff a user can't see. If you link it with a nextcloud you should do it with an user with proper rights.

Reply:
I agree it should be as it is - .with_user but no sudo